### PR TITLE
Added saving and loading of tournament participants

### LINKED ISSUE #69: Tournament participants
- in `api/create_tournament`the participants should be saved to the database
- in `src/tournament` only the participants should be placed in generated matches

### PR DESCRIPTION
**Pull request by AlgorithmicEntropy**
Fixed tournament mode enum with 

```SQL
ALTER table Tournament modify column `mode` enum('DEATHMATCH', 'SWISS', 'KO')
```

Closes #69

### CODE REVIEW

File: api/api.js [hunk 1]
```diff
@@ -64,6 +64,11 @@ function init (app) {
       }
 
       case 'createTournament': {
+        try {
+          require('../src/account_util').require_logged_in_user(req, res)
```
Comment (DavidRisch) [inline api/api.js]: This function is not intended for use in the api, redirecting only makes sense for get request. The previous implementation (`if (user === null)...`) was correct

File: src/tournament.js [hunk 2]
```diff
@@ -1,22 +1,60 @@
...
+    while (userPair = cmb.next()) { // eslint-disable-line no-cond-assign, no-undef
+      const round = require('../src/db/round').create(tournament.id)
+      const match = dbMatch.create(tournament.groupId, round.id, tournament.id)
+      match.addUser(userPair[0], 0) // eslint-disable-line no-undef
+      match.addUser(userPair[1], 1) // eslint-disable-line no-undef
+    }
+  } else if (matchMode === '2v2') {
+    // check length
+    if (participants % 4 !== 0) {
+      // TODO error handling
```
Comment (DavidRisch) [inline src/tournament.js:35]: This implies the code cannot handle a some number of participants. Please make an issue describing the current limitations. Some should be told to the user, some should be fixed in code

File: src/tournament.js [hunk 3]
```diff
@@ -34,10 +72,12 @@ function addRound (tournament) {
     for (let i = 0; i + 1 < remainingUsers.length; i += 2) {
       const userA = remainingUsers[i]
       const userB = remainingUsers[i + 1]
-      const match1 = dbMatch.create(tournament.groupId, roundId.id)
+      const match1 = dbMatch.create(tournament.groupId, roundId.id, tournament.id)
       match1.addUser(userA.id, 0)
       match1.addUser(userB.id, 1)
     }
+  } else if (tournament.mode === 'SWISS') {
+    // TODO implement swiss matchmaking
```
Comment (DavidRisch) [inline src/tournament.js:82]: Please make an issue for this

**Review by DavidRisch — CHANGES_REQUESTED** [review]
Some small things. I have not tested it yet.
The database has been updated

File: src/tournament.js (hunk 2 above)
Comment (AlgorithmicEntropy) [reply src/tournament.js:35]: Just realised that uneven numbers in 1v1 might also cause problems, will open a new issue describing both limitations

Comment (AlgorithmicEntropy) [reply src/tournament.js:35]: see #95

**Comment by AlgorithmicEntropy** [comment]
This pr might also  close #68, i did not realise it had its own issue

File: src/tournament.js (hunk 3 above)
Comment (AlgorithmicEntropy) [reply src/tournament.js:82]: see #96

**Comment by DavidRisch** [comment]
I was not able to fully test this due to #100

**Pull request merged**